rust/src/dirdiff.rs

```diff
@@ -131,34 +140,34 @@ fn canonicalize_name<'a>(prefix: Option<&str>, name: &'a str) -> Cow<'a, str> {
     }
 }
 
-fn diff_recurse(
-    prefix: Option<&str>,
-    src: &openat::Dir,
-    dest: &openat::Dir,
-    diff: &mut Diff,
-) -> Result<()> {
+fn diff_recurse(prefix: Option<&str>, src: &Dir, dest: &Dir, diff: &mut Diff) -> Result<()> {
     let list_prefix = prefix.unwrap_or(".");
-    for entry in src.list_dir(list_prefix)? {
+    for entry in src.read_dir(list_prefix).context("Reading dir")? {
         let entry = entry?;
-        let name = if let Some(name) = entry.file_name().to_str() {
+        let name = entry.file_name();
+        let name = if let Some(name) = name.to_str() {
             name
         } else {
             // For now ignore invalid UTF-8 names
             continue;
         };
         let path = canonicalize_name(prefix, name);
         let pathp = &*path;
-        let srctype = src.get_file_type(&entry)?;
-        let is_dir = srctype == openat::SimpleType::Dir;
+        let srctype = entry.file_type().context("Reading file type")?;
+        let is_dir = srctype.is_dir();
 
-        match dest.metadata_optional(pathp)? {
+        match dest
+            .symlink_metadata_optional(pathp)
+            .context("Reading dest path")?
```

Comment:
Hmm, we're not reading the contents though. Maybe instead "Reading dest file metadata" ?